src/node_crypto.cc

```diff
@@ -4297,11 +4393,35 @@ SignBase::Error Verify::VerifyFinal(const char* key_pem,
       goto exit;
   }
 
+  if (!EVP_DigestFinal_ex(&mdctx_, m, &m_len)) {
+    goto exit;
+  }
+
   fatal = false;
-  r = EVP_VerifyFinal(&mdctx_,
-                      reinterpret_cast<const unsigned char*>(sig),
-                      siglen,
-                      pkey);
+
+  pkctx = EVP_PKEY_CTX_new(pkey, nullptr);
+  if (!pkctx)
+    goto err;
+  if (EVP_PKEY_verify_init(pkctx) <= 0)
+    goto err;
+  if (pkey->type == EVP_PKEY_RSA || pkey->type == EVP_PKEY_RSA2) {
+    if (EVP_PKEY_CTX_set_rsa_padding(pkctx, padding) <= 0)
+      goto err;
+    if (padding == RSA_PKCS1_PSS_PADDING)
+      if (EVP_PKEY_CTX_set_rsa_pss_saltlen(pkctx, saltlen) <= 0)
+        goto err;
```

Comment:
style nit: can you add `{}` if the body of an `if` spans more than a single line?